src/cmap/worker.rs

```diff
@@ -462,41 +563,35 @@ async fn establish_connection(
     establish_result
 }
 
+/// Enum modeling the possible pool states as described in the CMAP spec.
+///
+/// The "closed" state is omitted here because the pool considered closed only
+/// once it goes out of scope and cannot be manually closed before then.
+#[derive(Debug)]
+enum PoolState {
+    /// Connections may not be checked out nor created in the background to satisfy minPoolSize.
+    Paused,
+
+    /// Pool is operational.
+    Ready,
+}
+
 /// Task to process by the worker.
 #[derive(Debug)]
 enum PoolTask {
```

Comment:
It was getting pretty annoying to maintain essentially two copies of this enum, so I just moved all the management request ones into a single case.